primer-rel8/src/Primer/Database/Rel8/Rel8Db.hs

```diff
@@ -139,6 +141,26 @@ runRel8DbT m = runReaderT (unRel8DbT m)
 -- Note that 'MonadLog' has a functional dependency from 'm' to 'l'.
 type MonadRel8Db m l = (ConvertLogMessage Rel8DbLogMessage l, MonadCatch m, MonadThrow m, MonadIO m, MonadLog (WithSeverity l) m)
 
+-- A helper function for creating a 'Session' from a database query.
+--
+-- Note that we have 2 choices here if the session name that was
+-- fetched from the database isn't a valid 'SessionName': either we
+-- can return a failure, or we can convert it to a valid
+-- 'SessionName'. This situation can only ever happen if we've made a
+-- mistake (e.g., we've changed the rules on what's a valid
+-- 'SessionName' and didn't run a migration), or if someone has edited
+-- the database directly, without going through the API. In either
+-- case, it would be bad if a student can't load their session just
+-- because a session name was invalid, so we opt for "convert it to a
+-- valid 'SessionName'".
+safeMkSession :: (SessionId, Text, UTCTime) -> Session
+safeMkSession (s, n, t) = Session s (safeMkSessionName n) (LastModified t)
+
+--
+-- It might be helpful if this function returned an indication of
+-- whether the original name was safe, but for now, we convert
+-- silently.
+
```

Comment:
This comment seems to have become detached from its code

Reply:
Fixed.